### PR TITLE
Add Storage Blob and Queue Service Connection from Docker Compose

### PR DESCRIPTION
**Pull request by eddumelendez**
# Description

Spring Boot also provides Service Connection from Docker Compose. This contribution adds support for Storage Blob and Storage Queue.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by rujche** [comment]
Hi, @eddumelendez 

Thanks for your contributing.

1. I updated the PR, please take a look.
2. Do you have plan about implementing `CosmosDockerComposeConnectionDetailsFactory`. 
 - For `spring-cloud-azure-docker-container`, it has dependency of `org.springframework.boot:spring-boot-test-containers` and `org.testcontainers:azure`.
   - `ContainerConnectionDetailsFactory` in `spring-boot-test-containers` has type parameter `C extends Container<?>`, so `CosmosDBEmulatorContainer` in `org.testcontainers.azure` can be used. 
   - In unit test, trust store related property like `javax.net.ssl.trustStore` can be built by `CosmosDBEmulatorContainer`.
 - For `spring-cloud-azure-docker-compose`, now it doesn't have any dependency about `testcontainer`. Seems `xxx-docker-compose` is not related with `testcontainer` related dependencies, it's not right to add `org.testcontainers.azure` dependency in `spring-cloud-azure-docker-compose`.

**Comment by eddumelendez** [comment]
Hi @rujche 

Thanks for updating the PR.

`CosmosDBEmulatorContainer` is already supported in `spring-cloud-azure-testcontainers`. See [CosmosContainerConnectionDetailsFactory](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/spring-cloud-azure-testcontainers/src/main/java/com/azure/spring/cloud/testcontainers/implementation/service/connection/cosmos/CosmosContainerConnectionDetailsFactory.java)

Regarding to docker-compose not sure yet how to handle the certificate.

**Comment by rujche** [comment]
Hi, @saragluna , please help to review this PR. Let's support Storage Blob and Queue first. Support Cosmos in another PR.

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    <developerConnection>scm:git:ssh://git@github.com:Azure/azure-sdk-for-java.git</developerConnection>
+    <url>https://github.com/Azure/azure-sdk-for-java</url>
+  </scm>
+  <issueManagement>
+    <system>GitHub</system>
+    <url>https://github.com/Azure/azure-sdk-for-java/issues</url>
+  </issueManagement>
+  <properties>
+    <!-- Enables fail on deprecated API usage. -->
+    <compiler.failondeprecatedstatus/>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-docker-compose/pom.xml:37]: maybe we shouldn't add this in our pom file

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  <dependencies>
+    <dependency>
+      <groupId>com.azure.spring</groupId>
+      <artifactId>spring-cloud-azure-autoconfigure</artifactId>
+      <version>5.15.0-beta.1</version>  <!-- {x-version-update;com.azure.spring:spring-cloud-azure-autoconfigure;current} -->
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-spring-data-cosmos</artifactId>
+      <version>5.15.0-beta.1</version> <!-- {x-version-update;com.azure:azure-spring-data-cosmos;current} -->
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-docker-compose/pom.xml:49]: can this be optional?

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    </dependency>
+
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-storage-blob</artifactId>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-docker-compose/pom.xml:54]: same for storage

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>com.github.spotbugs</groupId>
+        <artifactId>spotbugs-maven-plugin</artifactId>
+        <version>4.8.2.0</version> <!-- {x-version-update;springboot3_com.github.spotbugs:spotbugs-maven-plugin;external_dependency} -->
+        <configuration>
+          <skip>true</skip>
+        </configuration>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-docker-compose/pom.xml:112]: We should remove these

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml (hunk 1 above)
Comment (rujche) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml:37]: Why not? I just keep align with other Spring Cloud Azure modules. Here is the screenshot:
image

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml (hunk 2 above)
Comment (rujche) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml]: Updated.

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml (hunk 3 above)
Comment (rujche) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml:48]: "Optional" added.

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml (hunk 4 above)
Comment (rujche) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml:108]: Why? IMU, we should keep align with other Spring Cloud Azure modules. Here is the screenshot:
image

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  </issueManagement>
+  <properties>
+    <!-- Enables fail on deprecated API usage. -->
+    <compiler.failondeprecatedstatus/>
+    <jacoco.skip>true</jacoco.skip>
+  </properties>
+  <dependencies>
+    <dependency>
+      <groupId>com.azure.spring</groupId>
+      <artifactId>spring-cloud-azure-autoconfigure</artifactId>
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-docker-compose/pom.xml:43]: Should this one also be optional?

**Review by saragluna — APPROVED** [review]

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml (hunk 4 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml:109]: We will fix this in another PR

File: sdk/spring/spring-cloud-azure-docker-compose/pom.xml (hunk 1 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-docker-compose/pom.xml:37]: Synced with the author, and it's by design.

**Comment by rujche** [comment]
> Hi @rujche
> 
> Thanks for updating the PR.
> 
> `CosmosDBEmulatorContainer` is already supported in `spring-cloud-azure-testcontainers`. See [CosmosContainerConnectionDetailsFactory](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/spring-cloud-azure-testcontainers/src/main/java/com/azure/spring/cloud/testcontainers/implementation/service/connection/cosmos/CosmosContainerConnectionDetailsFactory.java)
> 
> Regarding to docker-compose not sure yet how to handle the certificate.

Link to related test PR: https://github.com/Azure/azure-sdk-for-java/pull/47958

**Pull request merged**